app/models/git_repository.rb

```diff
@@ -226,9 +226,16 @@ def worktree_params
         params[:password] = auth.password
       end
     end
+    params[:proxy_url] = proxy_url if Settings.git_repository_proxy.host
     params
   end
 
+  def proxy_url
+    uri_opts = Settings.git_repository_proxy.to_h.slice(:host, :port, :scheme, :path)
+    uri_opts[:path] ||= "/"
```

Comment:
You may want to add

```ruby
uri_opts[:port] &&= uri_opts[:port].to_i
```

We've hit it in the past where someone enters a value in the UI and it makes its way here as a String.

Reply:
Looks like this works fine:

```
irb(main):004:0> URI::Generic.build(:host => "example.com", :port => "5678")
=> #<URI::Generic //example.com:5678>
```

Reply:
Must have changed or perhaps it was a different method...I'll have to see if I can find that.